### PR TITLE
[BUG FIX] Escapes all single quotes from environment variables for bash scripts to run successfully.

### PR DESCRIPTION
**Pull request by Xheno**
### Checklist

- [*] Adds replace method to escape single quotes from container definition environment variables.
- [*] Includes container definition with single quote and test to ensure the feature works.

### Motivation, issues

This PR is a bug fix for issue #18 
The bug would be triggered when environment variables with single quotes are include in the container_env_vars say through autogenerated passwords or so. This would make the bash scripts handling the deployment exit midway the process. 

### Description
This PR will use the replace method to escape all single quotes in the container environment variables.

### CODE REVIEW

**Comment by lokst** [comment]
Thanks @Xheno , I'll test this and get back to you!

File: src/python/update_container_definitions.py [hunk 1]
```diff
@@ -26,7 +26,7 @@ def run(previous_task_definition, container_image_name_updates, container_env_va
 
     # Expected format: container=...,name=...,value=...,container=...,name=...,value=
     try:
-        env_kv_pairs = container_env_var_updates.split(',')
+        env_kv_pairs = container_env_var_updates.replace("'", "\\'").split(',')
```
Comment (lokst) [inline src/python/update_container_definitions.py]: I tried testing with task definitions containing single quotes (https://github.com/CircleCI-Public/aws-ecs-orb/commit/9f6b5686eafae70824ff9fe5b3a65921c0200332) and found that the integration tests had failed (https://circleci.com/gh/CircleCI-Public/aws-ecs-orb/2085) It looks like there might have been issues with sourcing the resultant bash script after the escaping.

Comment (Xheno) [reply src/python/update_container_definitions.py]: Let me dig deeper into this

**Comment by Xheno** [comment]
> @Xheno Could you take a look at my comment? Thanks!

Hey @lokst sorry for the delay, the bug really did become a lot more than I expected. Please run through and let me know if there is more to do if the integration tests fail.

**Comment by lokst** [comment]
Thanks @Xheno for making the modifications! 🙂 Unfortunately I think there is still an issue when testing with a single quote: https://github.com/CircleCI-Public/aws-ecs-orb/commit/148d6ff2a866644aba8d26fcea811fa18a564fa7

Job: https://circleci.com/gh/CircleCI-Public/aws-ecs-orb/2247

**Comment by Xheno** [comment]
> Thanks @Xheno for making the modifications! 🙂 Unfortunately I think there is still an issue when testing with a single quote: [148d6ff](https://github.com/CircleCI-Public/aws-ecs-orb/commit/148d6ff2a866644aba8d26fcea811fa18a564fa7)
> 
> Job: https://circleci.com/gh/CircleCI-Public/aws-ecs-orb/2247

Hey @lokst after replicating the problem in bash I figured it needed to be done in bash. Please review the new changes.

**Comment by lokst** [comment]
Thanks @Xheno , I'll check this out and get back to you!

File: src/python/tests/test_update_container_definitions.py [hunk 2]
```diff
@@ -33,7 +33,7 @@ def test_invalid_image_update_param_container(self):
         self.assertRaises(ValueError,
                           run, TestContainerDefinitionsUpdate.task_dfn_multi_containers,
                           'acontainer=web,container=timer,image-and-tag=golang', '')
-
+
```
Comment (lokst) [inline src/python/tests/test_update_container_definitions.py]: Could you remove the extra whitespace as it causes a lint error?

File: src/python/update_container_definitions.py [hunk 3]
```diff
@@ -1,6 +1,7 @@
 from __future__ import absolute_import
 import sys
 import json
+import re
```
Comment (lokst) [inline src/python/update_container_definitions.py]: Is this import still required?

File: src/orb.yml.hbs [hunk 4]
```diff
@@ -635,6 +635,9 @@ commands:
 
             # Prepare container definitions
             CONTAINER_DEFS=$(python $UPDATE_CONTAINER_DEFS_SCRIPT_FILE "$PREVIOUS_TASK_DEFINITION" "$CONTAINER_IMAGE_NAME_UPDATES" "$CONTAINER_ENV_VAR_UPDATES")
+            
+            # Escape single quotes from environment variables for BASH_ENV
+            CLEANED_CONTAINER_DEFS=$(sed -r "s/'+/'\\\''/g" \<<< $CONTAINER_DEFS)
```
Comment (lokst) [inline src/orb.yml.hbs]: What do you think of `CLEANED_CONTAINER_DEFS=$(echo $CONTAINER_DEFS | sed -E "s/'+/'\\\''/g")` for greater compatibility with different `sed` distributions and POSIX standards?

This solution seems to work really well except for one case, which is, when `CONTAINER_DEFS` contains more than one single quote adjacent to each other. It seems that the replacement will result in only one single quote.

Here is how I tested this:

```
circleci@d6984d82a16b:/$ CONTAINER_DEFS="just test''ing"
circleci@d6984d82a16b:/$ CLEANED_CONTAINER_DEFS=$(echo $CONTAINER_DEFS | sed -E "s/'+/'\\\''/g")
circleci@d6984d82a16b:/$ echo "export CCI_ORB_AWS_ECS_CONTAINER_DEFS='${CLEANED_CONTAINER_DEFS}'" >> /tmp/myscript.sh
circleci@d6984d82a16b:/$ source /tmp/myscript.sh
circleci@d6984d82a16b:/$ echo $CCI_ORB_AWS_ECS_CONTAINER_DEFS
just test'ing
```

File: src/python/update_container_definitions.py (hunk 3 above)
Comment (Xheno) [reply src/python/update_container_definitions.py]: No I don't think so thank you for this reminder

File: src/python/tests/test_update_container_definitions.py (hunk 2 above)
Comment (Xheno) [reply src/python/tests/test_update_container_definitions.py]: Will do

File: src/orb.yml.hbs (hunk 4 above)
Comment (Xheno) [reply src/orb.yml.hbs]: The suggestion you have provided works best. Let me update.

**Comment by lokst** [comment]
Thanks @Xheno, I'll review it and get back to you!

**Pull request merged**